File: MovieMania/MovieMania/APIController.m
//
//  APIController.m
//  MovieMania
//
//  Created by Isaiah Khan on 11/12/15.
//  Copyright © 2015 The Iron Yard. All rights reserved.
//

#import "APIController.h"

@implementation APIController

-(id)init//:(APIControllerProtocol *)delegate
{
    self = [super init];
    if (self)
    {
        // _delegator = delegate;

    }
    return self;
}

-(void)searchInOMDBFor:(NSString *)searchTerm
{
    NSString *urlString = [NSString stringWithFormat:@"Pedro, url goes here!",searchTerm];
    NSURL *url = [NSURL URLWithString:urlString];
    
    NSURLSessionConfiguration *configuration = [NSURLSessionConfiguration defaultSessionConfiguration];
    NSURLSession *session = [NSURLSession sessionWithConfiguration:configuration delegate:self delegateQueue:[NSOperationQueue mainQueue]];
    
    NSURLSessionDataTask *dataTask = [session dataTaskWithURL:url];
    
    [dataTask resume];
    
}

-(NSDictionary *)paresJSON:(NSData *)dictionary
{
    
}


@end

File: GitHubFriends/GitHubFriends/APIController.swift
//
//  APIController.swift
//  GitHubFriends
//
//  Created by Isaiah Khan on 10/27/15.
//  Copyright © 2015 The Iron Yard. All rights reserved.
//

import Foundation

class APIController
{
    var delegate: APIControllerProtocol
    
    init(delegate: APIControllerProtocol)
    {
        self.delegate = delegate 
    }
    
    func searchGitHubFor(userName: String)
    {
        if let escapedSearchTerm = userName.stringByAddingPercentEncodingWithAllowedCharacters(NSCharacterSet.alphanumericCharacterSet())
        {
            let url = NSURL(string: "https://api.github.com/users/embryoconcepts")
            let session = NSURLSession.sharedSession()
            let task = session.dataTaskWithURL(url!, completionHandler: {data, response, error -> Void in
                print("Task completed")
                if error != nil
                {
                    print(error!.localizedDescription)
                }
                else
                {
                    if let dictionary = self.parseJSON(data!)
                    {
                        if let gitHubUser: NSDictionary = dictionary["embryoconcepts"] as? NSDictionary
                        {
                            self.delegate.didReceiveAPIResults(embryoconcepts)
                        }
                    }
                }
            })
            task.resume()
        }
    }
    
    func parseJSON(data: NSData) -> NSDictionary?
    {
        do
        {
            let dictionary: NSDictionary! = try NSJSONSerialization.JSONObjectWithData(data, options: []) as! NSDictionary
            
            return dictionary
        }
        catch let error as NSError
        {
            print(error)
            return nil
        }
    }
    
    
    
    
    
    
}